### PR TITLE
feat: add devcontainer

### PR DESCRIPTION
**Pull request by caproven**
Adds devcontainer. Uses a debian base image with the following tools:
* Go
* PostgreSQL client (psql)
* redpanda client (rpk)
* Docker, socket from host
* Git

Would appreciate others testing this. From inside the devcontainer, should be able to run EPR with just `go run .`. If there's other utils that'd be useful then please share, and let me know if I should doc what all is available to users.

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM mcr.microsoft.com/devcontainers/go:1-1.21-bookworm
+
+RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
+    && apt-get -y install --no-install-recommends postgresql-client
+
+RUN curl -LO https://github.com/redpanda-data/redpanda/releases/download/v23.2.7/rpk-linux-amd64.zip \
+    && unzip rpk-linux-amd64.zip \
```
Comment (xbcsmith) [inline .devcontainer/Dockerfile:7]: Are we not going to validate the checksum here script kiddie?

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+FROM mcr.microsoft.com/devcontainers/go:1-1.21-bookworm
+
+RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
+    && apt-get -y install --no-install-recommends postgresql-client
```
Comment (xbcsmith) [inline .devcontainer/Dockerfile:4]: Do we want some other tools in here like `jq`, `golangci-lint`, and `python3.11`?
I use a pile of node based linting tools as well so maybe we add them?
Not saying to add all this stuff but here is my Fedora dev box guide as an example. https://github.com/xbcsmith/xbcsmith/blob/main/docs/fedora_38_dev_box.md

File: .devcontainer/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,91 @@
+version: '3.8'
```
Comment (xbcsmith) [inline .devcontainer/docker-compose.yml:1]: We now have versions of this file scattered all over the repo. Should we consolidate them all into one place (one source of the truth)?

Comment (caproven) [reply .devcontainer/docker-compose.yml:1]: Will work on this, good idea.

Comment (caproven) [reply .devcontainer/docker-compose.yml]: Consolidated. Noticing further duplication of doc content that could use revisiting but that's outside scope of this PR.

File: .devcontainer/Dockerfile (hunk 2 above)
Comment (caproven) [reply .devcontainer/Dockerfile:4]: To my surprise, this image already includes python (have to use `python3` explicitly). Let me know if there are any additional tools we should add.

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (caproven) [reply .devcontainer/Dockerfile:13]: Added checksum validation, scripting this is new to me so let me know if I missed the mark.

File: .devcontainer/Dockerfile (hunk 2 above)
Comment (xbcsmith) [reply .devcontainer/Dockerfile:4]: Lets start here

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (xbcsmith) [reply .devcontainer/Dockerfile:13]: LGTM

**Review by xbcsmith — APPROVED** [review]

**Pull request merged**